### PR TITLE
[ENH] Gpx browser

### PR DESCRIPTION
**Pull request by rksh**
Want access to logged routes? Here you go!
Note: option only appears if route logging is on.
Not applicable for FOSS release.

### CODE REVIEW

**Review by bobzilladev — APPROVED** [review]
Wow, you've been busy. All looks cogent to me, except the recyclerview stuff which I assume is Dark Alchemy that just must be done.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java [hunk 1]
```diff
@@ -997,166 +995,4 @@ protected void onProgressUpdate(Integer... values) {
             }
         }
     }
-    class AsyncGpxExportTask extends AsyncTask<Long, Integer, String> {
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java:1000]: mmm refactorsticking

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/util/AsyncGpxExportTask.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            //}
+            final double lat = cursor.getDouble(0);
+            final double lon = cursor.getDouble(1);
+            final long time = cursor.getLong(2);
+
+            writer.append("<trkpt lat=\"").append(String.valueOf(lat)).append("\" lon=\"")
+                    .append(String.valueOf(lon)).append("\"><time>").append(
+                    dateFormat.format(new Date(time))).append("</time></trkpt>\n");
+            lineCount++;
+            if (totalCount == 0) {
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/util/AsyncGpxExportTask.java:142]: i know this is just a refactor move, but that seems like a strange case, ne?

**Review by uhtu — APPROVED** [review]

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java (hunk 1 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java:1000]: now to move all the other asyncTasks out

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/util/AsyncGpxExportTask.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/util/AsyncGpxExportTask.java:142]: probably belt and suspenders. this one will need a hard look to make sure it works right everywhere.

**Pull request merged**